gnosis/cowswap
# 162/use is order unfillable

# Summary

Fixes #162 

Checking whether orders are unfillable

# To Test

1. On rinkeby (because it's easier to move the price of a fake token), pick GNO-WETH pair
2. Make sure you have about 5ETH worth of rinkeby GNO. During testing it was enough
3. Sell the equivalent of 2.5 WETH in GNO for WETH.
4. Right after, place a second order doing the same thing
5. Wait until one of the orders is matched
- [ ] The second order will no longer be "matchable", because the price should have moved against you (this depends on the pair liquidity, so YMMV)
- [ ] You should see a (very ugly) message under activity details stating that the price is out of range
![screenshot_2021-08-06_16-19-43](https://user-images.githubusercontent.com/43217/128579624-ba110db1-0e5a-48f2-9c14-77a18e589583.png)
- [ ] If you look at the console, you should see the pending order has a percentage greater than 1%, that represents the different between the first number (order price) to the second number (current price)
![screenshot_2021-08-06_16-29-03](https://user-images.githubusercontent.com/43217/128579833-9ccc243a-1dad-41d1-af9d-52390a1bb92f.png)



# Background

# Out of scope
- Make it look nice (Michel is working on that)
- Check against user's balance
